### ISSUE TITLE
Number.cmd should provide formatting options for output numbers.

### ISSUE THREAD

**Issue by timlg07** [timlg07/Number.cmd#23]

**Comment by timlg07** [comment]
Possible syntax for output formatting:

```cmd
Number x = 1 / 3 format:".3"
:: x = 0.333
Number x = 4 / 10 format:"1,2"
:: x = 0,40
Number x = 22 * 23 format:"1.2"
:: x = 5.06E2
```

**Comment by timlg07** [comment]
**A more accurate definition of the format syntax:**
  
"_x_._y_" with _x_ being the number of digits before and _y_ the number of digits after the decimal point. The precision in this case is always fixed and defined as: _precision_ = _x_ + _y_. This definition overwrites a manually given precision. The exponent will be adjusted to meet the format requirements. If the exponent is zero, it is omitted.
  
"_x_." with _x_ being the number of digits before the decimal point. In this case, all operations will be performed with default precision. The number of digits after the floating point is not limited and is used for all _actual-precision_ - _x_ remaining digits. In case no digits are leftover, which happens if and only if _actual-precision_ = _x_, the floating-point is omitted. The exponent will still be adjusted to meet the requirements of the format.
  
"._y_" with _y_ being the number of digits after the decimal point. Analogical to the previous case, the default precision behavior is used; all remaining digits will now be placed before the decimal point and the exponent will get adjusted as necessary. The floating-point won't get omitted in any case though: If _actual-precision_ = _y_ a leading 0 followed by the point will be added.
  
"." does not limit the amount of digits at all. The default precision behavior will be used and the digits will be split so that the exponent is zero and can be omitted completely. => **This is the only format, that will never contain an exponent**. 

If you add a plus sign `+` as first character of the format string, the sign is always shown, even when it might be redundant. This makes it easier for you to process the output, as you can always expect the first character to be the sign (Even if the result is zero).
  
Notes:
- All those formats can also have a different 'floating-point' delimiter like a comma.  
- Additional to the format, you can also specify the precision manually.

**Comment by timlg07** [comment]
Format parameter parser finished: #32.

**Comment by timlg07** [comment]
- [x] Formats like `0.0` should throw a warning and be ignored.

**Comment by timlg07** [comment]
Formatting algorithm implemented: #33

**Comment by timlg07** [comment]
Plus sign option implemented: #34 #36

**Comment by timlg07** [comment]
Basic automated tests written: #37
